### ISSUE TITLE
Wrong assumption about missing '-' arg if `pipingStdin` detected

### ISSUE THREAD

**Issue by aleksandr-vin** [mikefarah/yq#1115]
**Describe the bug**
When `yq` is placed inside a bigger block where other commands read from stdin which is piped to that whole block, `yq` is egoistically reading the stdin.

The commit 703418d0c4a54c05693dab294f6336ce79015692 adds "-" to the args when `pipingStdin` is detected.

Version of yq: 4.20.1
Operating system: mac
Installed via: homebrew

**Command**

Problem appears when used in blocks, so the minimal script that reproduces the problem:

```bash
#!/usr/bin/env bash
set -e
set -x

cat >foo.json <<EOF
{}
EOF

ls foo*
printf "1\n2\n" | while read -r f
do
    yq -P '.' "foo.json" >"foo-$f.yaml"
done
find . -type f -depth 1 -name 'foo-*.yaml' -print -exec cat {} ";"
```

**Actual behavior**

The `read` of the `while` block is intended to read `1` and `2` on each iteration of `while` but in practice there is nothing left for `read` after `yq` finished his one call, and `2` can be found in `foo-1.yaml`:

```
+ cat
+ ls foo.json
foo.json
+ printf '1\n2\n'
+ read -r f
+ yq -P . foo.json
+ read -r f
+ find . -type f -depth 1 -name 'foo-*.yaml' -print -exec cat '{}' ';'
./foo-1.yaml
{}
---
2
```

**Expected behavior**

```
+ cat
+ ls foo.json
foo.json
+ printf '1\n2\n'
+ read -r f
+ yq -P . foo.json
+ read -r f
+ yq -P . foo.json
+ read -r f
+ find . -type f -depth 1 -name 'foo-*.yaml' -print -exec cat '{}' ';'
./foo-1.yaml
{}
./foo-2.yaml
{}
```

**Workaround**

In such cases when `yq` is involved in a block of commands which has its stdin piped (but not for `yq`) the egoism of `yq` can be fed with `</dev/null`. So the line in the before mentioned script would be:

```
  yq -P '.' </dev/null "foo.json" >"foo-$f.yaml"
```

**Comment by mikefarah** [comment]
Yeah I can replicate that - but not actually sure how to fix it.

Any golang experts? I'm doing this to detect if there's stdin:

```go
stat, _ := os.Stdin.Stat()
pipingStdIn := (stat.Mode() & os.ModeCharDevice) == 0
```

which from what I read is the correct way, but yes it appears to be too eager :/

**Comment by aleksandr-vin** [comment]
Maybe adding `-` only if no files are on the args list?

**Comment by mikefarah** [comment]
...that's a great idea!
________________________________

**Comment by mikefarah** [comment]
Fixed in 4.20.2
